src/java/org/jivesoftware/openfire/plugin/util/cluster/ClusterPacketRouter.java

```diff
@@ -40,9 +41,11 @@
     public boolean routePacket(byte[] nodeID, JID receipient, Packet packet) {
         // Send the packet to the specified node and let the remote node deliver the packet to the recipient
         try {
+            if (!ClusterManager.isClusterMember(nodeID))
+                return false;
             CacheFactory.doClusterTask(new RemotePacketExecution(receipient, packet), nodeID);
             return true;
-        } catch (IllegalStateException  e) {
+        } catch (IllegalArgumentException | IllegalStateException  e) {
```

Comment:
I wonder if this should simply catch `java.lang.Exception`? I can't see a good reason for letting this fall through to the caller.

Reply:
done